### PR TITLE
[RFC] Extend Threat Fieldset - Stage 2 Proposal

### PR DESCRIPTION
**Pull request by peasead**
- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)? Yes
- Have you followed the [contributor guidelines](https://github.com/elastic/ecs/blob/master/CONTRIBUTING.md)? Yes
- For proposing substantial changes or additions to the schema, have you reviewed the [RFC process](https://github.com/elastic/ecs/blob/master/rfcs/README.md)? Yes
- If submitting code/script changes, have you verified all tests pass locally using `make test`? Yes
- If submitting schema/fields updates, have you generated new artifacts by running `make` and committed those changes? Yes
- Is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed. Yes
- Have you added an entry to the [CHANGELOG.next.md](https://github.com/elastic/ecs/blob/master/CHANGELOG.next.md)? Yes

[Preview of the markdown proposal doc](https://github.com/peasead/ecs/blob/extend-threat-stage-2/rfcs/text/0018-extend-threat-group-software.md)

## Stage 2 Criteria

- [x] Opened pull request for this draft revising the existing proposal
- [x] Completed field definitions
- [x] Included a real world example source document
- [x] Identifies scope of impact of changes to ingestion mechanisms (e.g. beats/logstash), usage mechanisms (e.g. Kibana applications, detections), and the ECS project (e.g. docs, tooling)
- [x] Subject matter experts weighed in on technical utility of field definitions in the pull request

### CODE REVIEW

**Comment by ebeahan** [comment]
Noted from reviewing the field definitions added in the last stage, do we want to keep `threat.software.platforms` plural? Based on the example, the field is expecting a single value:

https://github.com/elastic/ecs/blob/master/experimental/schemas/threat.yml#L216

**Comment by peasead** [comment]
> Noted from reviewing the field definitions added in the last stage, do we want to keep `threat.software.platforms` plural? Based on the example, the field is expecting a single value:
> 
> https://github.com/elastic/ecs/blob/master/experimental/schemas/threat.yml#L216

Thanks Eric.

Yes, platforms should be plural. I made the changes in the descriptions to be plural.

**Review by ebeahan — COMMENTED** [review]
Thanks for adjusting `platforms`. 👍 

I added a note about the using expected values for `software.platforms`, but I also think the same question around enumerating an expected value list for `software.type`.

File: rfcs/text/0018/threat.yml [hunk 1]
```diff
@@ -24,9 +24,9 @@
   - name: software.platforms
     level: extended
     type: keyword
-    short: Platform of the software.
+    short: platforms of the software.
     description: >
-      The platform of the software used by this threat to conduct behavior commonly modeled using MITRE ATT&CK®. While not required, you can use a MITRE ATT&CK® software platform.
+      The platforms of the software used by this threat to conduct behavior commonly modeled using MITRE ATT&CK®. While not required, you can use a MITRE ATT&CK® software platforms.
 
       Expected values
```
[the remark this answers is not in the document]
Comment (peasead) [reply rfcs/text/0018/threat.yml:31]: I think `PRE` may have been a stray copy/paste? I'll remove that. Good catch.

I checked all of the `threat.software.*` and `threat.group.*` examples and they are all source agnostic, we just used MITRE ATT&CK values for our examples. 

Are you suggesting we remove:
>...used by this threat to conduct behavior commonly modeled using MITRE ATT&CK®. While not required, you can use a MITRE ATT&CK®...

from all of the descriptions and simply use things like

>The id of the software used by this threat.`?

[the remark this answers is not in the document]
Comment (ebeahan) [reply rfcs/text/0018/threat.yml:31]: No, I think the description is great, and I don't have an issue mentioning MITRE ATT&CK.

My concern is the list under `Expected values`. The way it's presented, I'm worried users will interpret it as they _must_ use one of these values.

Maybe we swap `Expected` for `Recommended`?

```suggestion
      Recommended Values:
```

[the remark this answers is not in the document]
Comment (peasead) [reply rfcs/text/0018/threat.yml:31]: Oh, nice. I like that.

Committed.

**Review by ebeahan — APPROVED** [review]
LGTM 👍 

@devonakerr as the sponsor, would you also review?

**Comment by devonakerr** [comment]
Yessir.

**Review by devonakerr — APPROVED** [review]
LGTM - Andy, I appreciate the attention to detail.

**Comment by ebeahan** [comment]
Opened https://github.com/elastic/ecs/pull/1429 to set the correct advancement date on the RFC.

**Pull request merged**